src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt

```diff
@@ -268,6 +272,11 @@ private fun SpacingContext.blockMustBeMultiLine(): Boolean {
     }
 }
 
+private fun SpacingContext.needsBlankLineBetweenItems(): Boolean {
+    // Allow to keep consecutive runs of `use`, `const` or `static` items without blank lines
+    return !(elementType1 == elementType2 && elementType1 in ONE_LINE_ITEMS)
```

Comment:
I believe comment binders approach won't work, here is counterexample:
```rust
mod foo;
//! Where is your God now?
mod bar;
```

This guy has to be unbound comment element. We need to handle comments in formatter. As far as I remember PyCharm does this (and the `ncPsi` approach is inspired by it) but it's quite hard reference to grasp into imho because it's formatter looks like spaghetti code for me in some places :(